File: code/functions/pyprism/pyprism/data/tests/test_load_res.py
# -*- coding: utf-8 -*-
"""
@created: Jul 08 2021
@modified: Aug 06 2021

@author: Yoann Pradat

Tests for _load_res.py module.
"""

import os
from pyprism.data import load_resource

def test_load_resource():
    # CIViC
    df_gene = load_resource(database="civic", name="gene")

    # CGC
    df_gene = load_resource(database="cosmic", name="gene")

    # curated
    df_gene = load_resource(database="curated", name="gene")

    # Gencode
    df_gene = load_resource(database="gencode", name="gencode_v23")

    # OncoKB
    df_gene = load_resource(database="oncokb", name="gene")

File: code/functions/rprism/tests/testthat/test-load_res.R
# GENE LIST

test_that("load_resource() works", {
  skip_on_cran()

  # OncoKB
  cat("\n")
  df_gene <- load_resource(database="oncokb", name="gene")
  expect_true(is(df_gene, "data.frame"))


  # CIViC
  cat("\n")
  df_gene <- load_resource(database="civic", name="gene")
  expect_true(is(df_gene, "data.frame"))

  # curated
  cat("\n")
  df_gene <- load_resource(database="curated", name="gene")
  expect_true(is(df_gene, "data.frame"))

  # CGC
  cat("\n")
  df_gene <- load_resource(database="cosmic", name="gene")
  expect_true(is(df_gene, "data.frame"))

  # Gencode
  df_gene <- load_resource(database="gencode", name="gencode_v27")
  expect_true(is(df_gene, "data.frame"))

})
